modules/webbrowser/interaction/cefkeyboardmapping.cpp

```diff
@@ -0,0 +1,151 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2018 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/webbrowser/interaction/cefkeyboardmapping.h>
+
+#include <warn/push>
+#include <warn/ignore/all>
+#include "include/cef_base.h"
+#include <warn/pop>
+
+namespace inviwo {
+
+namespace cef {
+
+int mapKey(IvwKey key) {
+    switch (key) {
+        case IvwKey::Backspace:
+            return 8;
+        case IvwKey::Tab:
+            return 9;
+        case IvwKey::Enter:
+            return 13;
+        case IvwKey::Left:
+            return 37;
+        case IvwKey::Up:
+            return 38;
+        case IvwKey::Right:
+            return 39;
+        case IvwKey::Down:
+            return 40;
+        case IvwKey::Delete:
+            return 46;
+        case IvwKey::A:
+            return 97;
```

Comment:
what is the reason for capturing `a` and returning a lowercase `a`?
why not return `return 'a';` ?

Reply:
Cef keycodes are different. We need to investigate more because they don't translate properly at the moment. Also, we need to revamp our KeyEvent to capture more keys.